app/src/main/java/org/mozilla/focus/activity/MainActivity.java

```diff
@@ -701,7 +703,15 @@ private void onPreferenceClicked() {
 
     private void onExitClicked() {
         GeoPermissionCache.clear();
-        finish();
+        // if there's no private session, we just call finish. Otherwise, we ask PrivateModeActivity to drop
+        // BrowserFragment and show the home screen from there.
+        if (PrivateMode.hasPrivateSession(this)) {
```

Comment:
A privateModeActivity may be created anyway even if there is no private session. This implementation does not handle that case.